src/Button/Button.js

```diff
@@ -37,18 +41,85 @@ const Button = React.forwardRef(({
         className
     );
 
-    return (<button {...props} className={buttonClasses}
-        disabled={disabled} onClick={onClick}
-        ref={ref}
-        selected={selected}
-        type={typeAttr}>
-        {children}
-    </button>);
+    const ariaDisabled = props['aria-disabled'];
+    let disabledInAnyWay = disabled || ariaDisabled;
+
+    //if button is disabled and focusable, remove onClick handler
+    let onClickHandler = allowFocusOnDisable && disabledInAnyWay ? () => {} : onClick;
+
+    const ariaLiveId = shortId.generate();
+
+    const getDisabledProps = () => {
+        let disabledProps = {};
+        if (allowFocusOnDisable) {
+            disabledProps['aria-disabled'] = disabledInAnyWay;
+            disabledProps['aria-describedby'] = ariaLiveId;
+        } else {
+            disabledProps.disabled = disabledInAnyWay;
+        }
+        return disabledProps;
+    };
+
+    const liveMessageForScreenReaders = disabledInAnyWay ? disabledMessage : enabledMessage;
+
+    const renderButtonStateMessage = () => {
+        let content = null;
+        if (allowFocusOnDisable && disabledInAnyWay !== null) {
+            content = (
+                <p
+                    aria-live='assertive'
+                    className={'fd-button__instructions'}
+                    id={ariaLiveId}>
+                    {liveMessageForScreenReaders}
+                </p>)
+            ;
+        }
+        return content;
+    };
+    return (
+        <>
+            <button
+                {...props}
+                {...getDisabledProps()}
+                className={buttonClasses}
+                onClick={onClickHandler}
+                ref={ref}
+                selected={selected}
+                type={typeAttr}>
+                {children}
+            </button>
+            {renderButtonStateMessage()}
+        </>
+    );
 });
 
 Button.displayName = 'Button';
 
+const validateStateTransitionMessage = (props, propName) => {
+    if (props.allowFocusOnDisable && (props.disabled !== null || props['aria-disabled']) && !props[propName]) {
+        return new Error(`
+        Missing or invalid value for ${propName}.
+        If this button can recieve focus when disabled (i.e. allowFocusOnDisable prop is true),
+        you need to provide localized string values for the enabled and disabled states.
+        These strings are used to annouce the state change by assitive techonolgies.
```

Comment:
nit: `receive`, `announce`, `assistive technologies`
I know these are carryovers from nui-widgets 😅

Reply:
Didn't think to double check this haha. Will fix grammar.